edgedb/_cluster.py

```diff
@@ -136,29 +109,54 @@ def init(self):
 
         self._init()
 
-    def start(self, wait=60, **settings):
-        port = settings.pop('port', None) or self._port
-        if port == 'dynamic':
-            port = find_available_port()
+    def start(self, wait=60, *, port: int=None, **settings):
+        if port is None:
+            port = self._port
 
-        self._effective_port = port
+        if port == 0:
+            cmd_port = 'auto'
+        else:
+            cmd_port = str(port)
 
         extra_args = ['--{}={}'.format(k.replace('_', '-'), v)
                       for k, v in settings.items()]
-        extra_args.append('--port={}'.format(self._effective_port))
+        extra_args.append(f'--port={cmd_port}')
+        if port == 0:
+            extra_args.append('--echo-runtime-info')
 
         env = os.environ.copy()
         # Make sure the PYTHONPATH of _this_ process does
         # not interfere with the server's.
         env.pop('PYTHONPATH', None)
-
         if self._env:
             env.update(self._env)
 
+        stdout = subprocess.DEVNULL
+        if port == 0:
+            stdout = subprocess.PIPE
+
         self._daemon_process = subprocess.Popen(
             self._edgedb_cmd + extra_args,
-            stdout=sys.stdout, stderr=sys.stderr,
-            env=env, cwd=str(self._data_dir))
+            stdout=stdout,
+            stderr=subprocess.DEVNULL,
+            env=env,
+            universal_newlines=True,
+            cwd=str(self._data_dir)
+        )
+
+        if port == 0:
+            while 1:
```

Comment:
I would prefer there be two things:
1. Timeout
2. Spit edgedb output when timeout happens or when `json.loads` fails (including stderr). This was already a problem in JS (https://github.com/edgedb/edgedb-js/commit/3bdb7482bb4a9613c7f19e7fa3a162561d84b311)